dev/com.ibm.ws.ejbcontainer/src/com/ibm/ws/ejbcontainer/osgi/internal/NameSpaceBinderImpl.java

```diff
@@ -259,15 +260,15 @@ public void bindDefaultEJBRemote(EJBBinding bindingObject, HomeRecord hr) {
      * @param bindingName the JNDI binding name
      */
     private void bindLegacyRemoteBinding(EJBBinding bindingObject, HomeRecord hr, String bindingName) {
-        if (remoteRuntime != null) {
+        if (remoteRuntime != null && ejbRemoteRuntimeServiceRef.getReference() != null) {
```

Comment:
What is the scenario that can cause getReference() to be null?  Seems ejbRemoteRuntimeServiceRef.getService() would also be null; so wondering if we should stop caching it in remoteRuntime, and instead just call getService at the beginning of every function that needs it?  Or, can getService() return a non-null value when getReference() returns null?

Reply:
Would it be safer (less timing concern) with something like this:
```
        EJBRemoteRuntime remoteRuntime = ejbRemoteRuntimeServiceRef.getService();
        ServiceReference<EJBRemoteRuntime> remoteRuntimeRef = ejbRemoteRuntimeServiceRef.getReference();
        
        if (remoteRuntime != null && remoteRuntimeRef != null) {

            BundleContext bc = remoteRuntimeRef.getBundle().getBundleContext();
```

Reply:
the ServiceRef won't be null, but I kept forgetting that while investigating.

NamespaceBindleImpl init failing trace:
```
[7/26/20, 19:23:24:151 PDT] 00000029 id=00000000 com.ibm.ws.ejbcontainer.osgi.internal.NameSpaceBinderImpl    > <init> Entry  
                                                                                                               com.ibm.ws.ejbcontainer.osgi.internal.metadata.OSGiEJBModuleMetaDataImpl@b978b529[BindingNameTestApp#BindingNameEJB.jar]
                                                                                                               com.ibm.ws.ejbcontainer.osgi.internal.naming.EJBJavaColonNamingHelper@e42f6546
                                                                                                               com.ibm.ws.ejbcontainer.osgi.internal.naming.EJBLocalNamingHelperImpl@68e21ee0
                                                                                                               com.ibm.ws.ejbcontainer.osgi.internal.naming.LocalColonEJBNamingHelperImpl@7ba5c40a
                                                                                                               AtomicServiceReference[name=ejbRemoteRuntime,ctx=org.apache.felix.scr.impl.manager.ComponentContextImpl@73f649fd,ref=null,svc=null]
```
passing:
```
[7/30/20, 9:33:28:382 CDT] 0000005e id=00000000 com.ibm.ws.ejbcontainer.osgi.internal.NameSpaceBinderImpl    > <init> Entry  
                                                                                                               com.ibm.ws.ejbcontainer.osgi.internal.metadata.OSGiEJBModuleMetaDataImpl@b61d4416[ConfigTestsOtherTestApp#ConfigTestsOtherEJB.jar]
                                                                                                               com.ibm.ws.ejbcontainer.osgi.internal.naming.EJBJavaColonNamingHelper@3f5134dd
                                                                                                               com.ibm.ws.ejbcontainer.osgi.internal.naming.EJBLocalNamingHelperImpl@b8b59f55
                                                                                                               com.ibm.ws.ejbcontainer.osgi.internal.naming.LocalColonEJBNamingHelperImpl@34043eb2
                                                                                                               AtomicServiceReference[name=ejbRemoteRuntime,ctx=org.apache.felix.scr.impl.manager.ComponentContextImpl@b61c24f9,ref={com.ibm.ws.ejbcontainer.osgi.EJBRemoteRuntime, com.ibm.ws.transport.iiop.spi.RemoteObjectReplacer}={service.id=372, orbRef.target=(id=defaultOrb), service.bundleid=116, service.scope=bundle, LibertyProcess.target=(wlp.process.type=server), component.name=com.ibm.ws.ejbcontainer.remote.internal.EJBRemoteRuntimeImpl, component.id=288},svc=null]
```

Reply:
I did remove checking it being null but I do like getting the RemoteRuntime each time instead of caching it so I added that.

Reply:
actually `com.ibm.wsspi.kernel.service.utils.AtomicServiceReference.getService(boolean)`

throws IllegalStateException if the ref is null, so maybe I should be checking `ejbRemoteRuntimeServiceRef.getReference() != null`